File: product-service/src/main/java/com/intexsoft/webshop/productservice/dto/subcategory/SubcategoryCreateDto.java
package com.intexsoft.webshop.productservice.dto.subcategory;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.intexsoft.webshop.productservice.dto.attribute.AttributeCreateDto;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Positive;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class SubcategoryCreateDto {
    @NotBlank
    String name;
    @NotBlank
    String description;
    @Positive
    Long categoryId;
    @NotEmpty
    @JsonProperty("attributes")
    List<AttributeCreateDto> attributeCreateDtos;
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/dto/subcategory/SubcategoryCreateDto.kt
package com.intexsoft.webshop.productservicekt.dto.subcategory

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.intexsoft.webshop.productservicekt.dto.attribute.AttributeCreateDto
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.Positive
@JsonIgnoreProperties(ignoreUnknown = true)
data class SubcategoryCreateDto(
    @field:NotBlank
    val name: String?,
    @field:NotBlank
    val description: String?,
    @field:Positive
    val categoryId: Long,
    @JsonProperty("attributes")
    @field:Valid
    @field:NotEmpty
    val attributeCreateDtos: List<AttributeCreateDto>
)
